src/bam/record.rs

```diff
@@ -1779,4 +1818,60 @@ mod alignment_cigar_tests {
             );
         }
     }
+
+    #[test]
+    pub fn test_cigar_parsing_vs_regex() {
+        // parsing test cases
```

Comment:
Adding a test for a non-ASCII character in a CIGAR string:
```suggestion
    #[test]
    pub fn test_cigar_parsing_non_ascii_error() {
        let cigar_str = "43ጷ";
        expected_error = Err(Error::BamParseCigar {
                msg: "CIGAR string contained non-ASCII characters, which are not valid. Valid are [0-9MIDNSHP=X].".to_owned(),
            });

        result = CigarString::try_from(cigar_str).expect_err("This should return a BamParseCigar error, but somehow didn't.";
        assert_eq!(expected_error, result);
    }

    #[test]
    pub fn test_cigar_parsing() {
        // parsing test cases
```